File: code/intermediate/polymorphism/superKeywords/Child.js
class Parent {
  constructor() {
    console.log('Parent constructor called');
  }
  getMessage() {
    console.log('Message from parent');
  }
}

class Child extends Parent {
  constructor() {
    super();
    console.log('Child constructor called');
  }
  getMessage() {
    super.getMessage();
    console.log('Message from child');
  }
}

var child = new Child();
child.getMessage();

File: code/intermediate/polymorphism/superKeywords/Child.py
class Parent:
    def __init__(self):
        print("Parent constructor called")

    def printInfo(self):
        print("Method from parent called")


class Child(Parent):
    def __init__(self):
        # super().__init__()
        Parent.__init__(self)
        print("Constructor from child called")

    def printInfo(self):
        super().printInfo()
        print("Method from child called")


child = Child()
child.printInfo()
